src/dist/replication/lib/replica_stub.cpp

```diff
@@ -2540,5 +2618,28 @@ void replica_stub::on_notify_primary_split_catch_up(const notify_catch_up_reques
     }
 }
 
+void replica_stub::update_disk_holding_replicas()
+{
+    for (const auto &dir_node : _fs_manager._dir_nodes) {
+        // clear the holding_primary_replicas/holding_secondary_replicas and re-calculate it from
+        // holding_replicas
+        dir_node->holding_primary_replicas.clear();
+        dir_node->holding_secondary_replicas.clear();
+        for (const auto &holding_replicas : dir_node->holding_replicas) {
+            const std::set<dsn::gpid> &pids = holding_replicas.second;
+            for (const auto &pid : pids) {
+                replica_ptr replica = get_replica(pid);
+                if (replica != nullptr) {
+                    if (replica->status() == partition_status::PS_PRIMARY) {
+                        dir_node->holding_primary_replicas[holding_replicas.first].emplace(pid);
+                    } else if (replica->status() == partition_status::PS_SECONDARY) {
+                        dir_node->holding_secondary_replicas[holding_replicas.first].emplace(pid);
+                    }
+                }
```

Comment:
```suggestion
                if (!replica != nullptr) {
                    continue;
                }
                if (replica->status() == partition_status::PS_PRIMARY) {
                    dir_node->holding_primary_replicas[holding_replicas.first].emplace(pid);
                } else if (replica->status() == partition_status::PS_SECONDARY) {
                    dir_node->holding_secondary_replicas[holding_replicas.first].emplace(pid);
                }
```

Reply:
is
```
if (!replica != nullptr) {
                    continue;
                }
```
redundant code？I don't think it necessary.

Reply:
Of course it's necessary. `replica` can be null.

Reply:
I know, I means use "continue" is necessary?

Reply:
I think what @acelyc111 means is to reduce the embed levels of your codes.

Compare to:
```
if(true) {
  if(true) {
  else if(true) {
  }
}
```

apparently the following is better and clearer:

```
if(false) {
  continue;
}
if(true) {
} else if(true) {
}
```